### ISSUE TITLE
Make Ubuntu Travis happy again

### ISSUE THREAD

**Issue by milancurcic** [modern-fortran/tsunami#5]
Tsunami will build with `gfortran -fcoarray=single` when run on Travis CI but now that we're using `co_max` in `sync_edges`, the code doesn't build because `co_max` is not supported by Travis's gfortran. We need a more recent gfortran (7 or 8?) that supports collectives. See how Izaak or Jacob do this in their Travis files, I think it has something to do with importing some custom package repo in Travis Ubuntu.

@zbeekman Can you pls point me to your reference Travis config for Ubuntu with a recent gfortran, ideally 8?

**Comment by zbeekman** [comment]
You need to use the [toolchain-r-test PPA](https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test/+packages), which is [whitelisted by Travis-CI](https://github.com/travis-ci/apt-source-safelist/blob/master/ubuntu.json#L502). Note that the initial 8.1 release of GFortran was botched with a GFortran dev submitting a patch after the freeze that caused a severe regression, effectively breaking OpenCoarrays support in GFortran. However, I *think* the 8.1 version provided by the ubuntu-toolchain-r-test PPA has a patched 8.1 GFortran. If that doesn't work you'll need 7.x.

I use YAML anchors and references in the [OpenCoarrays .travis.yml](https://github.com/sourceryinstitute/OpenCoarrays/blob/master/.travis.yml) to avoid repeating myself so it may be a little hard to follow in some places if you're not used to the YAML reference/anchor syntax. [Lines 30-61](https://github.com/sourceryinstitute/OpenCoarrays/blob/master/.travis.yml#L30-L61) define two entries in the build matrix, one using GCC 6 and the other 7.

I'm slightly confused because I don't see 7 or 8 in the [package whitelists](https://github.com/travis-ci/apt-package-safelist) for Trusty or Precise, however, maybe all packages in whitelisted sources are themselves whitelisted? I know 7 works, I would guess 8 works too.

**Comment by milancurcic** [comment]
OK, great, thanks I will try that! I don't think the regression in 8.1 will be an issue right now since I just want to get this working with `gfortran -fcoarray=single`. Will worry about building OpenCoarray in Travis a bit later! :)

**Comment by zbeekman** [comment]
I'm pretty sure 8.1 is patched to fix the regression anyway... but if you see strange behavior let me know, and be aware that it could be that.
